data/src/main/java/io/gnosis/data/repositories/UnstoppableDomainsRepository.kt

```diff
@@ -0,0 +1,38 @@
+package io.gnosis.data.repositories
+
+import com.unstoppabledomains.config.network.model.Network
+import com.unstoppabledomains.exceptions.ns.NamingServiceException
+import com.unstoppabledomains.resolution.DomainResolution
+import com.unstoppabledomains.resolution.Resolution
+import com.unstoppabledomains.resolution.naming.service.NamingServiceType
+import io.gnosis.data.BuildConfig
+import pm.gnosis.model.Solidity
+import pm.gnosis.utils.asEthereumAddress
+import java.util.concurrent.Callable
+import java.util.concurrent.FutureTask
+
+class BackgroundTask : Callable<String> {
+    var address = ""
+    var domain: String;
+
+    constructor(domain: String)  {
+        this.domain = domain;
+    }
+
+    override fun call(): String {
+        val library: DomainResolution = Resolution.builder()
+                .providerUrl(NamingServiceType.CNS, BuildConfig.BLOCKCHAIN_NET_URL + BuildConfig.INFURA_API_KEY)
+                .build();
+        address = library.getAddress(domain, "eth");
+        return address
+    }
+}
+
+class UnstoppableDomainsRepository() {
```

Comment:
please inject DomainResolution as a dependency to the repository. Provide DomainResolution in ApplicationModule

Reply:
@elgatovital There has been a minor issue with injecting the library as a dependency to the repository. 

The library has been making a net_version call to the provider to get the correct network for self-configure which contract address to use. 
I couldn't found any way to make an async injection with dagger2, if anything I found this thread on [stackoverflow](https://stackoverflow.com/questions/57008801/how-combine-dagger2-and-coroutines) which is saying it is not possible to use suspend with the dagger, as it has no builtin suspend functionality. I am not very familiar with the dagger framework and not sure if there is any workaround for this.

Because of this, I parsed manually BuildConfig.BLOCKCHAIN_NET_URL to get the network from the first word of the URL assuming it is going to be infura. I found this very hacky and it probably won't work with other blockchain providers in case if you will switch to something else. 

Is there any better solution for getting the network?